src/marks/bollinger.js

```diff
@@ -0,0 +1,75 @@
+import {deviation, mean} from "d3";
+import {marks} from "../mark.js";
+import {map} from "../transforms/map.js";
+import {window} from "../transforms/window.js";
+import {areaX, areaY} from "./area.js";
+import {lineX, lineY} from "./line.js";
+import {identity} from "../options.js";
+
+export function bollingerBandX(
+  data,
+  {
+    x,
+    y,
+    n = 20,
+    k = 2,
+    color = "currentColor",
+    fill = color,
+    fillOpacity = 0.2,
+    stroke,
+    strokeOpacity,
+    strokeWidth,
+    ...options
+  } = {}
+) {
+  return marks(
+    areaX(data, bollingerX(n, k, {x, y1: y, y2: y, fill, fillOpacity, ...options})),
+    lineX(data, bollingerX(n, 0, {x, y, stroke, strokeOpacity, strokeWidth, ...options}))
+  );
+}
+
+export function bollingerBandY(
+  data,
+  {
+    x,
+    y,
+    n = 20,
+    k = 2,
+    color = "currentColor",
+    fill = color,
+    fillOpacity = 0.2,
+    stroke = color,
+    strokeOpacity,
+    strokeWidth,
+    ...options
+  } = {}
+) {
+  return marks(
+    areaY(data, bollingerY(n, k, {x, y1: y, y2: y, fill, fillOpacity, ...options})),
+    lineY(data, bollingerY(n, 0, {x, y, stroke, strokeOpacity, strokeWidth, ...options}))
+  );
+}
+
+export function bollingerX(n, k, options) {
+  let {x, x1, x2} = options;
+  if (x === undefined && x1 === undefined && x2 === undefined) options = {...options, x: (x = identity)};
+  const outputs = {};
+  if (x != null) outputs.x = bollinger(n, k);
+  if (x1 != null) outputs.x1 = bollinger(n, -k);
```

Comment:
By negating *k* here, bollingerX behaves like mapX, and can be used for either area or line!